tests/Doctrine/Tests/DBAL/Platforms/AbstractPlatformTestCase.php

```diff
@@ -632,17 +632,30 @@ public function testKeywordList() : void
         self::assertTrue($keywordList->isKeyword('table'));
     }
 
+    public function getAutoQuotedIdentifiers() {
+        return [
+            ['create'],
+            ['select'],
+            ['a name with spaces'],
+            ['a name with/slash'],
+            ['name-with-dash'],
+        ];
+    }
+
     /**
      * @group DBAL-374
+     * @group DBAL-2979
+     * @dataProvider getAutoQuotedIdentifiers
      */
-    public function testQuotedColumnInPrimaryKeyPropagation() : void
+    public function testAutoQuotedColumnInPrimaryKeyPropagation(string $identifier) : void
     {
         $table = new Table('`quoted`');
-        $table->addColumn('create', 'string');
-        $table->setPrimaryKey(['create']);
+        $table->addColumn($identifier, 'string');
+        $table->setPrimaryKey([$identifier]);
 
+        $expected = preg_replace('/create/', $identifier, $this->getQuotedColumnInPrimaryKeySQL());
```

Comment:
Not a huge fan of this… I would add a new argument defaulting to `create` to `getQuotedColumnInPrimaryKeySQL`.

Reply:
It's an abstract function with implementations across 4 or 7 files.  You intend for me to update them all?  There are also four or five functions that I will have to update this way.  I agree that my usage of preg_replace() could be a little fragile, so I don't mind.  I just wasn't sure that I should go so far with a first pass.

Reply:
> You intend for me to update them all? 

No I don't. That's why I said "defaulting to `create`"

Reply:
> You intend for me to update them all?

It will indeed be a lot of work, but I think this it will be for the best.

> I just wasn't sure that I should go so far with a first pass.

Yeah good call. Maybe let's wait for another contributor to chime in on this then? If they think we should merge this feature, and they are ok with my improvements for the tests, then you can proceed. @SenseException , maybe you could be of some advice on this?

Reply:
Incidentally, I am working on this now.  It's not that big a job.  I had considered it initially, but I was worried that it would make my original patch too invasive.

Reply:
Ok.  I started on the changes but I will put them on hold until I have your go-ahead, then.  Thanks.

Reply:
Yes, please don't use the preg_replace approach on this. I would appreciate when you do the improvement that @greg0ire was suggesting. Thank you for your effort in advance. :tulip:

Reply:
The patch is a lot bigger now, but there you go.  :)

Not sure of the etiquette of resolving conversations.  Should I do it to signify that I think I did what you asked, or should I wait for the original commenter (@greg0ire in this case) to resolve it to mark their satisfaction with the change?

Reply:
Thanks for asking, you should let the original commenter do the resolving.